src/rgw/rgw_sal.cc

```diff
@@ -296,8 +297,30 @@ StoreManager::Config StoreManager::get_config(bool admin, CephContext* cct)
 {
   StoreManager::Config cfg;
 
-  // Get the store backend
-  const auto& config_store = g_conf().get_val<std::string>("rgw_backend_store");
+  // Get the store backend and optional arguments
+  const std::string& config_store_str = g_conf().get_val<std::string>("rgw_backend_store");
+  std::string config_store{""};
+  std::string store_args{""};
+  std::regex store_regex("(\\w+)(\\w.*)*");
+  std::smatch store_match;
+  if (std::regex_match(config_store_str, store_match, store_regex)) {
+    if (store_match.size() > 2) {
+      store_args = store_match[2].str();
+    }
+      config_store = store_match[1].str();
```

Comment:
It looks like this should always match, but I want to make sure it will (I've never used std::regex before) since config_store is empty if the regex doesn't match.